compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1579,8 +1579,7 @@ private boolean workerExists(BType type, String workerName) {
 
     // Asynchronous Send Statement
     public void visit(BLangWorkerSend workerSendNode) {
-        BSymbol receiver = symResolver.lookupSymbol(env,
-                names.fromIdNode(workerSendNode.workerIdentifier), SymTag.VARIABLE);
+        BSymbol receiver = symResolver.lookupMainSpaceSymbol(env, names.fromIdNode(workerSendNode.workerIdentifier));
```

Comment:
Shouldn't we verify the resolved symbol is a variable here? I think we should.

Reply:
Changed